### PR TITLE
[#10245] UAT Round 2: Fix Help Page Links

### LINKED ISSUE #10003: Links in the Instructor's 'Getting Started' page do not work as expected
Links in the Instructor's 'Getting Started' page that should redirect to question cards on the Instructor's Help page do not work as expected.

- **Environment**: `master` branch at commit 44e0894. 

**Steps to reproduce**

1. Go to the `Getting Started` page for Instructors (the link to this page can be found at the top of the Instructor's Help page);

2. Try to follow any link within the steps that help instructors to get started using TEAMMATES.

**Expected behaviour**
Links should redirect to specific question cards on the Instructor Help page.
For example, clicking 'Enroll students in the course' link in the first 'Set up a course' step should redirect to 'How do I add students to a course?' question card on the Instructor Help page.

**Actual behaviour**
Links redirect to the top of the Instructor Help page instead of scrolling to the specific question cards.
teammates-issue

**Additional info**
This bug is probably due to the misconfiguration of fragment URLs (in particular, usage of `routerLink`s with `fragment`s). Additionally, links on the 'Getting started' page refer to the ids of the question cards on the Instructor Help page but these ids are actually missing.

### PR DESCRIPTION
**Pull request by t-cheepeng**
Part of #10245, Fixes #10003

Fixes
- Instructor help page: link fragments (the part at the end of links at the end that looks `#like-this`) are not respected. Proper link fragments should cause the page to scroll to the specified id.

gret

**Outline of Solution**
- Switch from `fragments` to `queryParams`
- Added scroll to question
- Added `expand` method to base class and implemented them in the different sections
- Refactored out the different id's used to control collapsing and links

### CODE REVIEW

**Comment by wkurniawan07** [comment]
There are links to the instructor help page from a few other pages too. Make sure to cover them as well.

**Comment by madanalogy** [comment]
Might be related to this issue as well https://github.com/TEAMMATES/teammates/issues/10003, could consider resolving that at the same time

File: src/web/app/pages-help/instructor-help-page/instructor-help-page.component.ts [hunk 1]
```diff
@@ -22,14 +24,22 @@ export class InstructorHelpPageComponent implements OnInit, AfterViewInit {
...
   ngAfterViewInit(): void {
-    this.route.fragment.subscribe((f: string) => {
-      this.scroll(f);
+    this.route.queryParams.subscribe((queryParam: Params) => {
+      if (queryParam.questionId) {
+        setTimeout(() => this.pageScrollService.scroll({
+          document: this.document,
+          scrollTarget: `#${queryParam.questionId}`,
+          scrollOffset: 70,
+        }), 500);
```
Comment (wkurniawan07) [inline src/web/app/pages-help/instructor-help-page/instructor-help-page.component.ts:41]: - The scrolling actions seems a bit too slow (based on the screenshot provided)
- Just scrolling is not sufficient; the relevant help section needs to be expanded

**Comment by t-cheepeng** [comment]
> The scrolling actions seems a bit too slow (based on the screenshot provided)

Have sped up the scroll speed. But there is still some delay due to the collapsable boxes needing to run through their collapse animation when the page load which should be resolved by #10281

**Comment by ccyccyccy** [comment]
> Have sped up the scroll speed. But there is still some delay due to the collapsable boxes needing to run through their collapse animation when the page load which should be resolved by #10281

I tried by merging into branch #10281. Can confirm will load faster.

**Comment by wkurniawan07** [comment]
I foresee merge conflicts with at least one of #10281 and #10276. Will defer my final review until both of them are merged and deconflicting (if necessary) is done.

**Comment by madanalogy** [comment]
> I foresee merge conflicts with at least one of #10281 and #10276. Will defer my final review until both of them are merged and deconflicting (if necessary) is done.

@t-cheepeng The two PRs have been merged, should be okay for you to start resolving conflicts now. Just re-request senior devs for review after you're done.

**Pull request merged**